stages/text-preprocessing/src/main/java/edu/kit/kastel/mcse/ardoco/core/text/providers/informants/corenlp/textprocessor/TextProcessorService.java

```diff
@@ -0,0 +1,79 @@
+/* Licensed under MIT 2023. */
+package edu.kit.kastel.mcse.ardoco.core.text.providers.informants.corenlp.textprocessor;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import edu.kit.kastel.mcse.ardoco.core.api.text.Text;
+import edu.kit.kastel.mcse.ardoco.core.text.providers.informants.corenlp.config.ConfigManager;
+import io.github.ardoco.textproviderjson.converter.DtoToObjectConverter;
+import io.github.ardoco.textproviderjson.converter.JsonConverter;
+import io.github.ardoco.textproviderjson.dto.TextDto;
+import io.github.ardoco.textproviderjson.error.InvalidJsonException;
+import io.github.ardoco.textproviderjson.error.NotConvertableException;
+
+/**
+ * This text processor processes texts by sending requests to a microservice, which provides text processing using CoreNLP.
+ */
+public class TextProcessorService {
+
+    /**
+     * processes and annotates a given text by sending requests to a microservice
+     * 
+     * @param inputText the input text
+     * @return the annotated text
+     */
+    public Text processText(String inputText) throws IOException, InvalidJsonException, NotConvertableException {
+        TextDto textDto;
+        String jsonText = sendCorenlpRequest(inputText);
+        textDto = JsonConverter.fromJsonString(jsonText);
+        return new DtoToObjectConverter().convertText(textDto);
+    }
+
+    private String sendCorenlpRequest(String inputText) throws IOException {
+        inputText = URLEncoder.encode(inputText, StandardCharsets.UTF_8);
+        String requestUrl = ConfigManager.getInstance().getProperty("microserviceUrl") + ConfigManager.getInstance().getProperty("corenlpService") + inputText;
+        return sendAuthenticatedGetRequest(requestUrl);
+    }
+
+    public String sendAuthenticatedGetRequest(String requestUrl) throws IOException {
+        String username = System.getenv("USERNAME");
+        String password = System.getenv("PASSWORD");
+        if (username == null || password == null) {
+            throw new IOException("Environment variables USERNAME and PASSWORD must be set.");
+        }
+        URL url = new URL(requestUrl);
```

Comment:
We could use `org.apache.hc.client5.http.impl.classic.HttpClients` here. I've used it in the diagram recognition module of ardoco.